arkouda/dataframe.py

```diff
@@ -2706,3 +2708,152 @@ def invert_permutation(perm):
     if (unique(perm).size != perm.size) and (perm.size != rng + 1):
         raise ValueError("The array is not a permutation.")
     return coargsort([perm, arange(perm.size)])
+
+
+@typechecked
+def inner_join_merge(left: DataFrame,
+                     right: DataFrame,
+                     on: str
+                     ) -> DataFrame:
+    """
+    Utilizes the ak.join.inner_join function to return an ak
+    DataFrame object containing only rows that are in both
+    the left and right Dataframes, (based on the "on" param),
+    as well as their associated values.
+
+    Parameters
+    ----------
+    left : DataFrame
+        The Left DataFrame to be joined
+    right : DataFrame
+        The Right DataFrame to be joined
+    on : str
+        The name of the DataFrame column the join is being
+        performed on
+
+    Returns
+    -------
+    ij_ak_df : DataFrame
+        Inner-Joined Arkouda DataFrame
+    """
+
+    ij = inner_join(left[on], right[on])
+
+    left_cols = left.columns.copy()
+    left_cols.remove(on)
+    right_cols = right.columns.copy()
+    right_cols.remove(on)
+
+    new_dict = {on: left[on][ij[0]]}
+
+    for col in left_cols:
+        new_dict[col] = left[col][ij[0]]
+    for col in right_cols:
+        new_dict[col] = right[col][ij[1]]
```

Comment:
let's say we have `col_i` which is in both `left.columns` and `right.columns`, but `col_i` is not in `on`. What the expected behavior for `col_i` in the inner joined return df?

Right now we are setting it equal to whatever values the `right` df has when indexed/permuted to meet the inner join condition (since we set it equal to left first and the overwrite it with right). I just want to make sure that's what we are intending to happen

If we are down to just always take `col_i` to be the right df, then we could have some unnecessary work since we first set `new_dict['col_i']` equal to `left['col_i][ij[0]]` then overwrite it with `right['col_i'][ij[1]]`. This could be quite a lot of unused indexing especially if `(left.columns ∩ right.columns) - on` is quite large

We can avoid this by doing something like `left_cols = set(left_cols) - set(right_cols)`. Hopefully this makes sense, if not we can jump on call to talk through it

Reply:
This is a great question, and one I hadn't accounted for.

The pandas behavior is to retain both columns and add a suffix to the end of each to indicate which is which (default is `_x` for left and `_y` for right), see pic below. This should be fairly straightforward to add

![image](https://github.com/Bears-R-Us/arkouda/assets/32350869/0ead4209-d8ba-4751-bf81-ccc2587d12b3)

Reply:
Addressed with [this commit](https://github.com/Bears-R-Us/arkouda/pull/2781/commits/243ea8966be66aa47355ef2df5828163f7a67f48)

Reply:
Nice!! We should prob add these arguments to the other functions as well